### ISSUE TITLE
Sync Settings type cast errors are still present

### ISSUE THREAD

**Issue by YoshiRulz** [TASVideos/BizHawk#2561]
The play movie dialog is still able to cause bad things to happen in `RomLoader` (at least it doesn't cause crashes). For example, Gambatte isn't available in the Linux port. If a movie using Gambatte is played back, then the following exception dialogs appear (after the expected `EntryPointNotFoundException`):
```
System.ArgumentException: Object of type 'BizHawk.Emulation.Cores.Nintendo.Gameboy.Gameboy+GambatteSyncSettings' cannot be converted to type 'BizHawk.Emulation.Cores.Nintendo.GBHawk.GBHawk+GBSyncSettings'.
  at System.RuntimeType.CheckValue (System.Object value, System.Reflection.Binder binder, System.Globalization.CultureInfo culture, System.Reflection.BindingFlags invokeAttr) [0x00071] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Reflection.RuntimeMethodInfo.ConvertValues (System.Reflection.Binder binder, System.Object[] args, System.Reflection.ParameterInfo[] pinfo, System.Globalization.CultureInfo culture, System.Reflection.BindingFlags invokeAttr) [0x00069] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Reflection.RuntimeConstructorInfo.DoInvoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00017] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Reflection.RuntimeConstructorInfo.Invoke (System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Reflection.ConstructorInfo.Invoke (System.Object[] parameters) [0x00000] in <671ef0784b04474c83bb500a849492f2>:0 
  at BizHawk.Emulation.Cores.CoreInventory+Core.CreateUsingLegacyConstructorParameters (BizHawk.Emulation.Cores.ICoreInventoryParameters cip) [0x000ee] in <da0a6f217ca84b3eb1ce3cfffe9d39d4>:0 
  at BizHawk.Emulation.Cores.CoreInventory+Core.Create (BizHawk.Emulation.Cores.ICoreInventoryParameters cip) [0x0000a] in <da0a6f217ca84b3eb1ce3cfffe9d39d4>:0 
  at BizHawk.Client.Common.RomLoader.MakeCoreFromCoreInventory (BizHawk.Client.Common.RomLoader+CoreInventoryParameters cip) [0x0009a] in <66323406191444a3bf04cff86520a845>:0 
```
then:
```
System.ArgumentException: Object of type 'BizHawk.Emulation.Cores.Nintendo.Gameboy.Gameboy+GambatteSyncSettings' cannot be converted to type 'BizHawk.Emulation.Cores.Consoles.Nintendo.Gameboy.Sameboy+SyncSettings'.
  at System.RuntimeType.CheckValue (System.Object value, System.Reflection.Binder binder, System.Globalization.CultureInfo culture, System.Reflection.BindingFlags invokeAttr) [0x00071] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Reflection.RuntimeMethodInfo.ConvertValues (System.Reflection.Binder binder, System.Object[] args, System.Reflection.ParameterInfo[] pinfo, System.Globalization.CultureInfo culture, System.Reflection.BindingFlags invokeAttr) [0x00069] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Reflection.RuntimeConstructorInfo.DoInvoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00017] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Reflection.RuntimeConstructorInfo.Invoke (System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Reflection.ConstructorInfo.Invoke (System.Object[] parameters) [0x00000] in <671ef0784b04474c83bb500a849492f2>:0 
  at BizHawk.Emulation.Cores.CoreInventory+Core.CreateUsingLegacyConstructorParameters (BizHawk.Emulation.Cores.ICoreInventoryParameters cip) [0x000ee] in <da0a6f217ca84b3eb1ce3cfffe9d39d4>:0 
  at BizHawk.Emulation.Cores.CoreInventory+Core.Create (BizHawk.Emulation.Cores.ICoreInventoryParameters cip) [0x0000a] in <da0a6f217ca84b3eb1ce3cfffe9d39d4>:0 
  at BizHawk.Client.Common.RomLoader.MakeCoreFromCoreInventory (BizHawk.Client.Common.RomLoader+CoreInventoryParameters cip) [0x0009a] in <66323406191444a3bf04cff86520a845>:0 
```
then the first again.

Repro'd on Linux using 2.5.2 and 47c98c6a5. ~~It will probably repro on Windows~~ Repro's on Windows if you remove `libgambatte.dll`.

I'm curious if this would ever happen on Windows without deleting libraries. It would need to be with NES, SNES, GB/C or PCE (the systems with multiple cores), as the play movie dialog fails fast when the loaded core's systems don't include the movie's system.

see also: #1185 (I thought there were more like this but I can't find them, guess they were in IRC or Discord), #2246

**Comment by nattthebear** [comment]
Also see https://github.com/TASVideos/BizHawk/issues/2246.

What exactly do you want to happen here?  A gambatte movie needs gambatte to play back.  If gambatte is not available, the movie will not play back, so what can we improve?

**Comment by YoshiRulz** [comment]
Having only one error dialog is the ideal UX I think. That's not the point though, there shouldn't be anywhere in the codebase where `GambatteSyncSettings` is cast to `GBSyncSettings`.

**Comment by nattthebear** [comment]
> Having only one error dialog is the ideal UX I think.

I just tried reproing this on Windows and did get the same errors, but you omitted an important fact:  First, a dialog message is shown that says `Multiple cores failed to load the rom:"`, and then each of the individual dialogs.

So it's not like the thing is spraying you with random messages, it really did try the movie in gbhawk, sameboy, and subgbhawk and they all failed, and the messaging is indicating that.

> That's not the point though, there shouldn't be anywhere in the codebase where GambatteSyncSettings is cast to GBSyncSettings.

RomLoader can detect this and provide a different error message, I guess?

**Comment by YoshiRulz** [comment]
Calling this fixed after 39c0011d0 and c0b6bf61b.
